packages/data/src/registry.js

```diff
@@ -117,11 +119,9 @@ export function createRegistry( storeConfigs = {} ) {
 	 */
 	function registerResolvers( reducerKey, newResolvers ) {
 		namespaces[ reducerKey ].resolvers = mapValues( newResolvers, ( resolver ) => {
-			if ( ! resolver.fulfill ) {
-				resolver = { fulfill: resolver };
-			}
-
-			return resolver;
+			const normalizedResolver = pick( resolver, [ 'shouldInvalidate', 'isFulfilled' ] );
```

Comment:
The whitelist seems a maintenance burden and makes the middleware less insulated. Is there something we could do with spread to keep all properties during normalization? 

I'll think on it.

Reply:
Maybe?

```js
const { fulfill = resolver } = resolver;
return { ...resolver, fulfill };
```

I'd worried about the behavior of spread on functions, but it seems smart enough:

![image](https://user-images.githubusercontent.com/1779930/47502487-f6c87a00-d835-11e8-8d32-2a2b3112bd84.png)